### ISSUE TITLE
Tailwind styles do not render on nx library components

### ISSUE THREAD

**Issue by brumargues** [nrwl/nx#13250]
### Current Behavior

I've encountered weird behaviour when loading a tailwind tsx component from the lib workspace. 
I created the component in the lib folder and imported it inside an app component, and when I ran `serve`, the tailwind styles were not being applied to that specific component (./lib).

I tried many solutions, like adding a tailwind.config file to the ./lib folder or loading the tailwind styles in a CSS inside the components folder, nothing seemed to work.

So, by "mistake", I created an empty file, with the same component's name in the ./lib folder, inside the folder of the main component in the apps folder, and the after recompiling, the styles loaded. And now, even after I deleted the empty file, the styles are still recognised (cache, I guess?)

Example of the file structure with the working solution:
apps
    |_ myApp
            |_ components
                         |_ comp-x.tsx // imports component "comp-child-x"
                                   |_ comp-child-x.tsx // empty file
libs
  |_ components
               |_ comp-child-x.tsx

### Expected Behavior

It is expected that the method `createGlobPatternsForDependencies` from the tailwind config would catch all the files.

### Steps to Reproduce

1. Create a component in the ./libs folder
2. Import that component inside a component in the ./apps folder
3. Create a new empty file inside the ./apps folder with the same naas of the the component in the ./libs folder
4. nx serve app

### Nx Report

```shell
Node : 16.17.1
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 14.8.4
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.1.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.1.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 14.8.4
   @nrwl/expo : Not Found
   @nrwl/express : 14.8.6
   @nrwl/jest : 15.1.0
   @nrwl/js : 15.1.0
   @nrwl/linter : 14.8.4
   @nrwl/nest : Not Found
   @nrwl/next : 15.1.0
   @nrwl/node : 14.8.4
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.1.0
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.1.0
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.1.0
   @nrwl/web : 15.1.0
   @nrwl/webpack : 15.1.0
   @nrwl/workspace : 14.8.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by AgentEnder** [comment]
Hey @brumargues , it may be unrelated, but some of your @nrwl/* packages don't line up with the version of `nx` you have installed. Aside from `@nrwl/nx-cloud`, all packages under the nrwl scope should generally be set up with the same version. 

Can you correct this and check that the issue still reproduces? An easy way to fix this would be to run `nx migrate {v}`, where `v` is the version of nx you have installed. Alternatively, updating to latest with `nx migrate latest` would also work.

**Comment by brumargues** [comment]
Hi @AgentEnder, 

Thank you for your prompt reply.

After the process I described in my initial comment, and without running, `nx migrate latest`, all the files started to be indexed and recognised by the compiler.

So I've created a new project from scratch with similar configurations, and everything works without issues.

I can't replicate my initial issue, so I think we can close this issue.

------
Also, I've run `nx migrate latest` on my project and found no issues.
